Frontend/ui-customer/src/Screens/TicketBookingFlow/MetroMyTickets/Transformer.purs

```diff
@@ -22,20 +22,32 @@ import Services.API
 import Data.Array
 import Data.Maybe
 import Engineering.Helpers.Commons
-
+import Data.Function.Uncurried as DFU
+import JBridge as JB
 
 metroTicketListApiToMyTicketsTransformer ::  (Array MetroTicketBookingStatus) -> MetroMyTicketsScreenState -> MetroMyTicketsScreenState 
 metroTicketListApiToMyTicketsTransformer ticketList state = 
   let 
-    activeTickets' = metroTicketCardTransformer $ filter (\ (MetroTicketBookingStatus ticket) -> (any (_ == ticket.status) ["CONFIRMED"])) ticketList
-    pastTickets' = metroTicketCardTransformer $ filter (\ (MetroTicketBookingStatus ticket) -> (any (_ == ticket.status) ["PAYMENT_PENDING", "FAILED"])) ticketList
+    activeTickets' = metroTicketCardTransformer $ filter (\ (MetroTicketBookingStatus bookingStatus) -> activeTicketEvaluator bookingStatus.status bookingStatus.tickets) ticketList
+    pastTickets' = metroTicketCardTransformer $ filter (\ (MetroTicketBookingStatus bookingStatus) -> pastTicketEvaluator bookingStatus.status bookingStatus.tickets) ticketList
   in
     state {
       data {
         activeTickets =  activeTickets'
       , pastTickets = pastTickets'
       }
     }
+  where 
+    activeTicketEvaluator :: String -> Array FRFSTicketAPI  -> Boolean
+    activeTicketEvaluator status tickets = 
+      let validTill = maybe "" (\(FRFSTicketAPI ticket) -> ticket.validTill) (head tickets)
+      in (any (_ == status) ["CONFIRMED", "PAYMENT_PENDING"]) && (not $ isTicketExpired validTill)
+
+    pastTicketEvaluator :: String -> Array FRFSTicketAPI -> Boolean
```

Comment:
@spravinkumar9952 sort tickets based on time is not handled

Reply:
coming in descending order from backend. so reversed the array